download-basket/src/main/java/downloadbasket/actions/GetFeatureForCropping.java

```diff
@@ -0,0 +1,86 @@
+package downloadbasket.actions;
+
+import downloadbasket.helpers.Helpers;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceIbatisImpl;
+import fi.nls.oskari.util.ResponseHelper;
+
+@OskariActionRoute("GetFeatureForCropping")
+public class GetFeatureForCropping extends ActionHandler {
+
+	private final Logger LOGGER = LogFactory.getLogger(GetFeatureForCropping.class);
+		
+    private static final String PARAM_LAYERS = "layers";
+    private static final String PARAM_X = "x";
+    private static final String PARAM_Y = "y";
+    private static final String PARAM_BBOX = "bbox";
+    private static final String PARAM_WIDTH = "width";
+    private static final String PARAM_HEIGHT = "height";
+    private static final String PARAM_SRS = "srs";
+    private static final String PARAM_URL = "url";
+    private static String FINAL_WMS_URL = "";
+
+	@Override
+    public void handleAction(final ActionParameters params) throws ActionException {
+		
+        final JSONArray data = new JSONArray();
+        
+		OskariLayerService mapLayerService = new OskariLayerServiceIbatisImpl();
```

Comment:
Initialize service on ActionHandler.init() (run once at server start) instead of handleAction() (run for each request)

Reply:
fixed